web/src/app/components/login-page/login-page.component.ts

```diff
@@ -54,32 +54,29 @@ export class LoginPageComponent implements OnInit {
 
   saveUser(user: User, provider: string) {
     this.userService.saveUser(user, provider)
-      .subscribe((userData) => {
-        this.saveUserSocialRecord();
-        this.saveUserSocialStats(userData, provider);
-      });
+      .pipe(
+        map(value => this.saveUserSocialStats(value, provider)),
+        map(() => this.saveUserSocialRecord()),
+        mergeAll()
+      )
+      .subscribe((res) => {
+        this.addProviderToUserSocial(res);
+      }, error => this.errorService.logError(error));
   }
 
+
   saveUserSocialRecord() {
-    this.userService.getUser()
-      .pipe(
-        map(value => this.addProviderToUserSocial(value)),
-        catchError(error => this.errorService.logError(error))
-      )
-      .subscribe();
+    return this.userService.getUser();
```

Comment:
`saveUserSocialRecord` doesn't seem to describe the methods actions with `getUser()`?